packages/jest-editor-support/src/ScriptParser.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const {babylonParser} = require('./parsers/BabylonParser');
+import type {Location} from './types';
+
+class Node {
+  start: Location;
+  end: Location;
+  file: string;
+}
+
+class Expect extends Node {}
+
+class ItBlock extends Node {
+  name: string;
+}
+
+type ParserReturn = {
+  itBlocks: Array<ItBlock>, 
+  expects: Array<Expect>
+}
+
+/**
+ * Converts the file into an AST, then passes out a
+ * collection of it and expects.
+ */
+function parse(file: string): ParserReturn {
+  if (file.endsWith('.js')) {

```

Comment:
files may end with `jsx` or some other extensions. I'm fine though to support js only for now. What do you think about changing this to check for `.ts` and if it isn't a `.ts`, we just try-catch with babylon?